program/src/instructions/mplx_core/sell_nft_token_pool.rs

```diff
@@ -216,197 +130,205 @@ pub fn process_sell_nft_token_pool_core<'info>(
+
+        let Fees {
+            taker_fee,
+            tamm_fee,
+            maker_broker_fee,
+            taker_broker_fee,
+        } = calc_taker_fees(current_price, MAKER_BROKER_PCT)?;
+
+        // No mm_fee for token pools.
+
+        // Validate asset account and determine if royalites need to be paid.
+        let royalties = validate_asset(
+            &ctx.accounts.asset.to_account_info(),
+            ctx.accounts
+                .collection
+                .as_ref()
+                .map(|a| a.to_account_info())
+                .as_ref(),
+        )?;
 
-    // Signer seeds for the pool account.
-    let signer_seeds: &[&[&[u8]]] = &[&[
-        b"pool",
-        owner_pubkey.as_ref(),
-        pool.pool_id.as_ref(),
-        &[pool.bump[0]],
-    ]];
-
-    // Transfer the NFT from the seller directly to the pool owner.
-    TransferV1CpiBuilder::new(&ctx.accounts.mpl_core_program)
-        .asset(&ctx.accounts.asset)
-        .authority(Some(&ctx.accounts.seller.to_account_info()))
-        .new_owner(&ctx.accounts.owner)
-        .payer(&ctx.accounts.seller)
-        .collection(ctx.accounts.collection.as_ref().map(|c| c.as_ref()))
-        .invoke()?;
-
-    /*  **Transfer Fees**
-    The sell price is the total price the seller receives for selling the NFT into the pool.
-
-    sell_price = current_price - taker_fee - creators_fee
-
-    taker_fee = tamm_fee + maker_broker_fee + taker_broker_fee
-
-    No mm_fee for token pools.
-
-    Fees are paid by deducting them from the current price, with the final remainder
-    then being sent to the seller.
-    */
-
-    // If the source funds are from a shared escrow account, we first transfer from there
-    // to the pool, to make payments cleaner. After this, we can always send from the pool
-    // so the logic is simpler.
-    if pool.shared_escrow != Pubkey::default() {
-        let incoming_shared_escrow = unwrap_opt!(
-            ctx.accounts.shared_escrow.as_ref(),
-            ErrorCode::BadSharedEscrow
-        )
-        .to_account_info();
+        let royalty_fee = if let Some(Royalties { basis_points, .. }) = royalties {
+            basis_points
+        } else {
+            0
+        };
 
-        let escrow_program = unwrap_opt!(
-            ctx.accounts.escrow_program.as_ref(),
-            ErrorCode::EscrowProgramNotSet
-        )
-        .to_account_info();
+        // No optional royalties.
+        let creators_fee = calc_creators_fee(royalty_fee, current_price, None, Some(100))?;
 
-        // Validate it's a valid escrow account.
-        assert_decode_margin_account(
-            &incoming_shared_escrow,
-            &ctx.accounts.owner.to_account_info(),
+        // For keeping track of current price + fees charged (computed dynamically)
+        // we do this before PriceMismatch for easy debugging eg if there's a lot of slippage
+        let event = TAmmEvent::BuySellEvent(BuySellEvent {
+            current_price,
```

Comment:
Intentional all of this stuff got moved around?

Reply:
Not sure what got moved around?